### PR TITLE
Fix for Policy Base Routing failure; Possible table ID collision.

### PR DESCRIPTION
**Pull request by ljkiraly**
Related issue: networkservicemesh/deployments-k8s/issues/9119

Added a mechanism to protect table ID selection from parallel runs.

### CODE REVIEW

**Comment by ljkiraly** [comment]
New solution based on this hint: https://github.com/networkservicemesh/sdk-kernel/pull/583#issuecomment-1559663683

File: pkg/kernel/networkservice/connectioncontextkernel/ipcontext/iprule/common.go [hunk 1]
```diff
@@ -54,52 +56,73 @@ func create(ctx context.Context, conn *networkservice.Connection, tableIDs *Map)
...
 			}
-			ps[tableID] = policy
-			tableIDs.Store(conn.GetId(), ps)
 		}
 	}
 	return nil
 }
 
+func addPolicy(ctx context.Context, netlinkHandle *netlink.Handle, policy *networkservice.PolicyRoute, l netlink.Link, ps policies, tableIDs *Map, tableID int, connID string, nsrtid NetnsRTableNextID, nsRTableNextIDToConnID *NetnsRTableNextIDToConnMap) error {
+	// If policy doesn't contain any route - add default
```
Comment (LionelJouin) [inline pkg/kernel/networkservice/connectioncontextkernel/ipcontext/iprule/common.go:106]:
```suggestion
	// release the lock
	nsRTableNextIDToConnID.Delete(nsrtid)
	// If policy doesn't contain any route - add default
```

If the routes / policy routes fails to be added, the lock will never be released (line 122)

**Review by denis-tingaikin — APPROVED** [review]
LGTM

@ljkiraly Could you apply fixes for the comment https://github.com/networkservicemesh/sdk-kernel/pull/588/files#r1227954152?

File: pkg/kernel/networkservice/connectioncontextkernel/ipcontext/iprule/common.go [hunk 2]
```diff
@@ -54,52 +56,75 @@ func create(ctx context.Context, conn *networkservice.Connection, tableIDs *Map)
...
+			return err
+		}
+	}
+	if err := ruleAdd(ctx, netlinkHandle, policy, tableID); err != nil {
+		return err
+	}
+	ps[tableID] = policy
+	tableIDs.Store(connID, ps)
+	// release the lock
+	nsRTableNextIDToConnID.Delete(nsrtid)
```
Comment (LionelJouin) [inline pkg/kernel/networkservice/connectioncontextkernel/ipcontext/iprule/common.go:124]:
```suggestion
```
This could be removed

**Review by LionelJouin — APPROVED** [review]

**Pull request merged**